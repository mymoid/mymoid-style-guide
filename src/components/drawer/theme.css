@import '../../colors.css';
@import '../../variables.css';
@import '../../media.css';
@import '../app_bar/config.css';
@import './config.css';

.wrapper {
  position: relative;
  z-index: var(--z-index-zero);
}

.drawer {
  background-color: var(--drawer-background-color);
  box-shadow: var(--shadow-2p);
  color: var(--drawer-text-color);
  display: block;
  height: 100vh;
  /*overflow-x: hidden;*/
  /*overflow-y: auto;*/
  pointer-events: none;
  position: fixed;
  top: var(--appbar-height); /*TODO: top: 0, */
  transform-style: preserve-3d;
  transition: transform var(--animation-duration) var(--animation-curve-default);
  transition-delay: 0s;
  width: var(--drawer-mobile-width);
  will-change: transform;

  @apply --reset;

  &.active {
    pointer-events: all;
    transform: translateX(0px);
    transition-delay: var(--animation-delay);
  }

  &.right {
    right: 0;

    &:not(.active) {
      pointer-events: all; /*TODO: remove this, may be need an collapse state  */
      transform: translateX(100%);
    }
  }

  &.left {
    left: 0;

    &:not(.active) {
      pointer-events: all; /*TODO: remove this, may be need an collapse state  */
      transform: translateX(
        calc(calc(-1 * 100%) + 60px)
      ); /*TODO: override in dashboard.css*/
    }
  }
}

@media screen and (--larger-than-xs-viewport) {
  .drawer {
    width: var(--drawer-desktop-width);
  }
}
