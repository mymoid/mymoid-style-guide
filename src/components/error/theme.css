@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.error {
  @apply --reset;

  font-family: Lato;
  font-size: var(--error-font-size);
  color: var(--error-color);
  letter-spacing: 0;
  padding: var(--error-padding);
}
