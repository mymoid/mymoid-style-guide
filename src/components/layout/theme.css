@import '../../colors.css';
@import '../../variables.css';
@import '../../media.css';
@import './config.css';

@import '../app_bar/config.css';
@import '../drawer/config.css';

:root {
  --layout-side-transition: all var(--animation-duration)
    var(--animation-curve-default);
}

.layout {
  align-items: stretch;
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100vh;
  min-width: 100%;
  position: relative;

  @apply --reset;
}

.panel {
  left: 0;
  position: absolute;
  right: 0;
  top: 0;

  &:not(.bodyScroll) {
    height: 100vh;
    max-height: 100vh;
    overflow-y: scroll;
  }
}

.navDrawerDrawer {
  z-index: var(--z-index-high);

  &.pinned {
    box-shadow: none;
    left: 0;
  }

  &.clipped {
    height: calc(100vh - var(--appbar-height));
    padding-top: calc(0.5 * var(--unit));
    top: var(--appbar-height);

    @media screen and (--xxs-viewport) and (--portrait) {
      height: calc(100vh - var(--appbar-height-m-portrait));
      top: var(--appbar-height-m-portrait);
    }

    @media screen and (--xs-viewport) and (--landscape) {
      height: calc(100vh - var(--appbar-height-m-landscape));
      top: var(--appbar-height-m-landscape);
    }
  }
}

.appbarInner,
.panel {
  transition: var(--layout-side-transition);
}

.appbarFixed {
  &.appbarAppBar {
    z-index: var(--z-index-high);
  }

  & .panel {
    top: var(--appbar-height);

    &:not(.bodyScroll) {
      height: calc(100vh - var(--appbar-height));
      max-height: calc(100vh - var(--appbar-height));
      overflow-y: scroll;
    }

    @media screen and (--xxs-viewport) and (--portrait) {
      top: var(--appbar-height-m-portrait);
    }

    @media screen and (--xs-viewport) and (--landscape) {
      top: var(--appbar-height-m-landscape);
    }
  }
}

.navDrawerPinned {
  & .appbarLeftIcon {
    display: none;
  }

  & .panel {
    left: var(--drawer-mobile-width);
  }

  &:not(.navDrawerClipped) {
    & .appbarAppBar {
      padding-left: calc(var(--drawer-mobile-width) + var(--appbar-h-padding));
    }
  }

  @media screen and (--larger-than-xs-viewport) {
    & .panel {
      left: var(--drawer-desktop-width);
    }

    &:not(.navDrawerClipped) {
      & .appbarAppBar {
        padding-left: calc(
          var(--drawer-desktop-width) + var(--appbar-h-padding)
        );
      }
    }
  }
}

.navDrawerClipped {
  & .navDrawerWrapper {
    position: relative;
    z-index: var(--z-index-normal);
  }
}

.panelFloatingCentered {
  margin-left: auto;
  position: relative;
  margin-right: auto;
  margin-top: calc(0.35 * var(--floating-panel-height));
  margin-bottom: calc(0.35 * var(--floating-panel-height));
}

.panelContent {
  background: var(--panel-content-background-color);
  box-shadow: 0 5px 40px 0 rgba(0, 0, 0, 0.31);
  border-radius: calc(0.1 * var(--unit));
}

.panelActions {
  align-items: center;
  display: flex;
  justify-content: center;
  padding: var(--panel-padding-sm);
  background: var(--panel-actions-background);
  opacity: var(--panel-actions-background-opacity);
  box-shadow: var(--panel-actions-box-shadow);

  & [data-mymoid='button'] {
    margin: 0 calc(var(--panel-padding-sm) / 2);
    min-width: 0;
    padding: 0 var(--panel-padding-sm);

    &:first-child {
      margin-left: 0;
    }

    &:last-child {
      margin-right: 0;
    }
  }
}

.panelTitle {
  align-items: center;
  display: flex;
  justify-content: center;
  padding: var(--panel-title-padding-sm);
}

@define-mixin sidebarIncrements $platform {
  & .sidebarDrawer {
    width: calc(var(--standard-increment-$(platform)) * $(increment));
  }

  &.sidebarPinned {
    & .panel {
      right: calc(var(--standard-increment-$(platform)) * $(increment));
    }

    &:not(.sidebarClipped) {
      & .appbarAppBar {
        padding-right: calc(
          var(--standard-increment-$(platform)) * $(increment) +
            var(--appbar-h-padding)
        );
      }
    }
  }
}

@each $increment in 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 {
  .sidebarWidth$(increment) {
    & .sidebarDrawer {
      width: 100%;
    }

    @media screen and (--larger-than-sm-viewport) {
      @mixin sidebarIncrements desktop;
    }
  }
}

@each $increment in 1, 2, 3, 4, 5, 6, 7, 8, 9 {
  .sidebarWidth$(increment) {
    @media screen and (--larger-than-xs-viewport) and (--landscape) {
      @mixin sidebarIncrements mobile;
    }

    @media screen and (--larger-than-xs-viewport) and (orientation: portrait) {
      @mixin sidebarIncrements desktop;
    }
  }
}

@each $increment in 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 {
  .sidebarWidth$(increment) {
    @media screen and (--larger-than-sm-tablet-viewport) {
      @mixin sidebarIncrements desktop;
    }
  }
}

@each $pct in 25, 33, 50, 66, 75, 100 {
  .sidebarWidth$(pct) {
    & .sidebarDrawer {
      width: 100%;
    }
  }
}

@media screen and (--larger-than-sm-tablet-viewport) {
  @each $pct in 25, 33, 50, 66, 75, 100 {
    .sidebarWidth$(pct) {
      & .panel {
        right: calc($(pct) * 1%);
      }

      & .sidebarDrawer {
        width: calc($(pct) * 1%);
      }

      &:not(.sidebarClipped) {
        & .appbarAppBar {
          padding-right: calc(calc($(pct) * 1%) + var(--appbar-h-padding));
        }
      }
    }
  }
}
