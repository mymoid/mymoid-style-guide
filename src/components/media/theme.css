@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.media {
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;

  &.wide,
  &.square {
    width: 100%;

    & .content {
      height: 100%;
      position: absolute;
    }

    & .content > iframe,
    & .content > video,
    & .content > img {
      max-width: 100%;
    }
  }

  &::after {
    content: '';
    display: block;
    height: 0;
  }

  &.wide::after {
    padding-top: 56.25%;
  }

  &.square::after {
    padding-top: 100%;
  }

  & .content {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    left: 0;
    overflow: hidden;
    position: relative;
    top: 0;
    width: 100%;
  }
}
