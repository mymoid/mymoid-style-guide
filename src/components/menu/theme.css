@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.iconMenu {
  display: inline-block;
  position: relative;
  text-align: center;

  @apply --reset;

  & .icon {
    cursor: pointer;
    /*TODO: refactor magic number to na*/
    height: 100%;
  }
}

.active {
  background-color: transparent;
}

.menu {
  display: inline-block;
  position: relative;

  &.topLeft {
    position: absolute;
    left: 0;
    top: 0;

    & > .outline {
      transform-origin: 0 0;
    }
  }

  &.topRight {
    position: absolute;
    left: 100%;
    top: 10px;

    & > .outline {
      transform-origin: 100% 0;
    }
  }

  &.bottomLeft {
    position: absolute;
    top: calc(100% - 1px);
    left: 0;

    & > .outline {
      transform-origin: 0 100%;
    }
  }

  &.bottomRight {
    position: absolute;
    /*TODO: refactor this*/
    top: calc(100% - 1px);
    right: 0;

    & > .outline {
      transform-origin: 100% 100%;
    }
  }

  &:not(.static) {
    pointer-events: none;
    z-index: var(--z-index-higher);

    & > .outline {
      opacity: 0;
      transform: scale(0);
      transition: transform var(--menu-expand-duration)
          var(--animation-curve-default),
        opacity var(--menu-fade-duration) var(--animation-curve-default);
      will-change: transform;
    }

    & > .menuInner {
      left: 0;
      margin: 0;
      opacity: 0;
      position: absolute;
      top: 0;
    }

    &.rippled:not(.active) {
      & > .outline {
        transition-delay: var(--menu-ripple-delay);
      }

      & > .menuInner {
        transition-delay: var(--menu-ripple-delay);
      }
    }

    &.active {
      pointer-events: all;

      & > .outline {
        opacity: 1;
        transform: scale(1);
      }

      & > .menuInner {
        opacity: 1;
        transition: opacity var(--menu-fade-duration)
            var(--animation-curve-default),
          clip var(--menu-expand-duration) var(--animation-curve-default);
      }
    }
  }
}

.outline {
  background-color: var(--menu-background-color);
  border-radius: var(--menu-outline-border-radius);
  box-shadow: var(--shadow-2p);
  display: block;
  left: 0;
  position: absolute;
  top: 0;
}

.menuInner {
  display: block;
  list-style: none;
  padding: var(--menu-padding);
  position: relative;
  text-align: left;
  white-space: nowrap;
}

.menuItem {
  align-items: center;
  color: var(--menu-item-color);
  display: flex;
  font-size: var(--menu-item-font-size);
  height: var(--menu-item-height);
  overflow: hidden;
  padding: 0 var(--menu-item-padding);
  position: relative;

  &:not(.disabled):hover {
    background-color: var(--menu-item-hover-background);
    cursor: pointer;
  }

  &.disabled {
    opacity: 0.5;
    pointer-events: none;
  }

  &.selected {
    background-color: var(--menu-item-selected-background);
    font-weight: 500;
  }

  & .ripple {
    color: var(--color-white); /*TODO: --color-text-secondary*/
  }

  & .icon {
    font-size: var(--menu-item-icon-font-size);
    width: var(--menu-item-icon-size);
    height: initial;
    font-family: Lato;
    color: #cdd8dd;
    letter-spacing: 0;
  }
}

.caption {
  flex-grow: 1;
  font-size: var(--font-size-normal);
}

.shortcut {
  margin-left: var(--menu-item-padding);
}

.menuDivider {
  background-color: var(--color-divider);
  border: 0;
  display: block;
  height: 1px;
  margin: var(--menu-divider-height) 0;
  outline: 0;
  padding: 0;
  width: 100%;
}

.profileTitle {
  align-items: center;
  display: flex;
  text-transform: none;
  text-align: left;

  & [data-mymoid='avatar'] {
    margin-right: calc(2.1 * var(--unit));
  }

  & .title {
    font-family: Lato;
    font-size: calc(2 * var(--unit));
    font-weight: 500;
    letter-spacing: 0.02em;
    line-height: 1;
    margin: 0;
    padding: 0;
  }

  & .subtitle {
    font-family: Lato;
    font-weight: 300;
    color: var(--menu-profile-subtitle-color);
    font-size: calc(1.2 * var(--unit));
    text-transform: uppercase;
    font-weight: 400;
    letter-spacing: 0;
    line-height: calc(2.3 * var(--unit));
    margin: 0;
    padding: 0;

    &.active {
      color: var(--menu-profile-subtitle-active-color);
    }
  }

  &.large {
    padding: var(--menu-profile-title-padding-lg)
      var(--menu-profile-title-padding)
      calc(var(--menu-profile-title-padding) - 0.2 * var(--unit));

    & .title {
      font-size: calc(2.4 * var(--unit));
      -moz-osx-font-smoothing: grayscale;
      font-weight: 400;
      line-height: 1.25;
    }
  }

  &.small {
    padding: var(--menu-profile-title-padding) 6px;

    & .title {
      letter-spacing: 0;
      font-size: calc(1.6 * var(--unit));
      line-height: 1.4;
      color: var(--menu-profile-title-color);

      &.active {
        color: var(--menu-profile-title-active-color);
      }
    }

    & .subtitle {
      letter-spacing: calc(0.07 * var(--unit));
      font-weight: 500;
      line-height: 2.3;
    }
  }
}
