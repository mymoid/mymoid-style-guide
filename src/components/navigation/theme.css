@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.horizontal {
  & > [data-mymoid='button'],
  & > [data-mymoid='link'] {
    display: inline-block;
    margin: 0 calc(var(--navigation-space) / 2);
  }
}

.vertical {
  & > [data-mymoid='button'],
  & > [data-mymoid='link'] {
    display: block;
    margin: calc(var(--navigation-space) / 2);
  }
}

.vertical,
.horizontal {
  padding: calc(var(--navigation-space) / 2);

  @apply --reset;

  & > [data-mymoid='link'] {
    color: var(--navigation-color);
  }
}
