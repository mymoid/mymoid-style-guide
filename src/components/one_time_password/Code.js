import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

class Code extends Component {
  constructor(props) {
    super(props);

    const { value, fields, type, isValid, disabled } = props;

    this.state = {
      value,
      fields,
      type,
      input: [],
      isValid,
      disabled,
      defaultInputStyle: {
        fontFamily: 'Lato',
        fontWeight: '300',
        fontSize: '48px',
        border: 'none',
        width: '43px',
        height: '43px',
        boxSizing: 'border-box'
      }
    };
    for (let i = 0; i < Number(this.state.fields); i += 1) {
      if (i < 32) {
        const changeValue = [...this.state.value][i] || '';
        this.state.input.push(changeValue);
      }
    }
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      isValid: nextProps.isValid,
      value: nextProps.value,
      disabled: nextProps.disabled
    });
  }

  onKeyDown(e) {
    const target = Number(e.target.id);
    const prevTarget = this.refs[target - 1]; // eslint-disable-line react/no-string-refs
    let input;
    let value;

    switch (e.keyCode) {
      case 8:
        e.preventDefault();
        this.refs[target].value = ''; // eslint-disable-line react/no-string-refs
        input = this.state.input.slice();
        input[target] = '';
        value = input.join('');

        this.setState({ value, input });
        // eslint-disable-next-line react/no-string-refs
        if (this.refs[target].value === '') {
          if (prevTarget) {
            prevTarget.focus();
            prevTarget.select();
          }
        }
        break;
      default:
        break;
    }
    if ('onChange' in this.props) {
      if (value) {
        this.props.onChange(value);
      }
    }
    this.handleTouch(value);
  }

  handleBlur(e) {
    this.handleTouch(e.target.value);
  }

  handleChange(e) {
    const target = Number(e.target.id);
    let value = String(e.target.value);

    if (value !== '') {
      if (this.state.type === 'number' || this.state.type === 'text') {
        if (!value.match(/(\d)/g)) {
          this.refs[target].value = ''; // eslint-disable-line react/no-string-refs
          return;
        }
        if (value.length > 1) {
          this.refs[target].value = value.slice(-1); // eslint-disable-line react/no-string-refs
        }
      }

      const input = this.state.input.slice();
      input[target] = this.refs[target].value; // eslint-disable-line react/no-string-refs
      value = input.join('');

      this.setState({ value, input });

      const newTarget = this.refs[target + 1]; // eslint-disable-line react/no-string-refs
      if (newTarget) {
        newTarget.focus();
        newTarget.select();
      }
    }
    if ('onChange' in this.props) {
      if (value) {
        this.props.onChange(value);
      }
    }
    this.handleTouch(value);
  }

  handleTouch(value) {
    const { touch, untouch, name } = this.props;

    if (typeof touch === 'function' && typeof untouch === 'function') {
      if (value === '') {
        touch(name);
      } else {
        untouch(name);
      }
    }
  }

  render() {
    const {
      className,
      style = {},
      inputStyle = {},
      inputStyleInvalid = {},
      theme,
      type
    } = this.props;
    const { disabled, input, isValid, defaultInputStyle } = this.state;
    const styles = {
      container: style,
      input: isValid ? inputStyle : inputStyleInvalid
    };

    Object.assign({}, styles.container, {
      display: 'inline-block'
    });

    if (!className && Object.keys(inputStyle).length === 0) {
      Object.assign(inputStyle, {
        ...defaultInputStyle,
        fontFamily: 'Lato',
        fontWeight: '300',
        fontSize: '48px',
        color: '#000000',
        letterSpacing: '8px',
        backgroundColor: 'rgb(250, 250, 250)',
        borderColor: 'lightgrey'
      });
    }

    if (!className && Object.keys(inputStyleInvalid).length === 0) {
      Object.assign(inputStyleInvalid, {
        ...defaultInputStyle,
        color: 'rgba(255, 0, 0, 100)'
      });
    }

    if (disabled) {
      Object.assign(styles.input, {
        cursor: 'not-allowed',
        color: 'lightgrey',
        borderColor: 'lightgrey',
        backgroundColor: '#efeff1'
      });
    }

    const classNames = classnames(theme.codeInput, this.props.className);

    return (
      <div className={classNames} style={styles.container}>
        {input.map((value, i) => (
          <input
            ref={i}
            id={i}
            autoFocus={i === 0 ? 'autoFocus' : ''}
            defaultValue={value}
            key={`input_${i}`} // eslint-disable-line
            type={type}
            min={0}
            max={9}
            maxLength={1}
            style={styles.input}
            autoComplete="off"
            onFocus={e => e.target.select(e)}
            onBlur={e => this.handleBlur(e)}
            onChange={e => this.handleChange(e)}
            onKeyDown={e => this.onKeyDown(e)}
            disabled={disabled}
            data-valid={isValid}
            placeholder="â€¢"
          />
        ))}
      </div>
    );
  }
}

Code.defaultProps = {
  isValid: true,
  disabled: false,
  fields: 4,
  value: '',
  type: 'text'
};

Code.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
  fields: PropTypes.number,
  inputStyle: PropTypes.object, // eslint-disable-line
  inputStyleInvalid: PropTypes.object, // eslint-disable-line
  isValid: PropTypes.bool,
  name: PropTypes.string,
  onChange: PropTypes.func,
  options: PropTypes.object, // eslint-disable-line
  style: PropTypes.object, // eslint-disable-line
  theme: PropTypes.shape({
    codeInput: PropTypes.string
  }),
  touch: PropTypes.func,
  type: PropTypes.oneOf(['text', 'number', 'password']),
  untouch: PropTypes.func,
  value: PropTypes.string
};

export { Code };
