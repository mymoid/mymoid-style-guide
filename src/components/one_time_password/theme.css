@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.codeInput {
  & > *:focus {
    outline: none;
  }

  & > *::-webkit-input-placeholder {
    font-family: Lato;
    font-weight: 300;
    font-size: var(--otp-input-font-size);
    color: var(--otp-input-placeholder-color);
    letter-spacing: 8px;
  }

  input[type='number']::-webkit-outer-spin-button,
  input[type='number']::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  input[type='text']::-webkit-input-placeholder {
    font-family: Lato;
    font-weight: 300;
    font-size: 48px;
    color: #d8d8d8;
    letter-spacing: 8px;
  }
  input[type='number'] {
    -moz-appearance: textfield;
  }
}
