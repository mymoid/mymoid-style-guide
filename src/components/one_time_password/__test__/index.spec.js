import React from 'react';
import Enzyme, { render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Code } from '../Code';
import theme from '../theme.css';

Enzyme.configure({ adapter: new Adapter() });

const numbers = '123456';

describe('Otp', () => {
  describe('#render', () => {
    it(`renders without error`, () => {
      const wrapper = shallow(<Code className={'code-class'} theme={theme} />);
      expect(wrapper).toMatchSnapshot();
    });

    it(`renders field without any values`, () => {
      const wrapper = shallow(<Code theme={theme} />);
      expect(wrapper.find('input').length).toEqual(4);
      expect(wrapper.state().fields).toEqual(4);
      expect(wrapper.state().value).toEqual('');
      expect(wrapper.state().type).toEqual('text');
      expect(wrapper.type()).toEqual('div');
    });

    it(`renders component with value: ${numbers}`, () => {
      const wrapper = render(
          <Code theme={theme} value={numbers} fields={6} type="number" />
        ),
        val = [];
      for (let i = 0; i < numbers.length; i += 1) {
        val.push(wrapper.find('input')[i].attribs.value);
      }
      expect(val.join('')).toEqual(numbers);
      expect(Number(val.join(''))).toBe(123456);
    });

    it(`mount component with props: "fields={6}"`, () => {
      const wrapper = shallow(<Code theme={theme} fields={6} />);
      expect(wrapper.find('input').length).toEqual(6);
    });

    it(`should have 4 input felds`, () => {
      const wrapper = shallow(<Code theme={theme} />);
      expect(wrapper.find('input').length).toEqual(4);
    });
  });
});
