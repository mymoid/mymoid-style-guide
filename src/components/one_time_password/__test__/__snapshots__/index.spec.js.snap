// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Otp #render renders without error 1`] = `
<div
  className="code-class"
  style={Object {}}
>
  <input
    autoComplete="off"
    autoFocus="autoFocus"
    data-valid={true}
    defaultValue=""
    disabled={false}
    id={0}
    key="input_0"
    max={9}
    maxLength={1}
    min={0}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="•"
    style={Object {}}
    type="text"
  />
  <input
    autoComplete="off"
    autoFocus=""
    data-valid={true}
    defaultValue=""
    disabled={false}
    id={1}
    key="input_1"
    max={9}
    maxLength={1}
    min={0}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="•"
    style={Object {}}
    type="text"
  />
  <input
    autoComplete="off"
    autoFocus=""
    data-valid={true}
    defaultValue=""
    disabled={false}
    id={2}
    key="input_2"
    max={9}
    maxLength={1}
    min={0}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="•"
    style={Object {}}
    type="text"
  />
  <input
    autoComplete="off"
    autoFocus=""
    data-valid={true}
    defaultValue=""
    disabled={false}
    id={3}
    key="input_3"
    max={9}
    maxLength={1}
    min={0}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="•"
    style={Object {}}
    type="text"
  />
</div>
`;
